### DISCUSSION TITLE
Using Azure.xx Baseline line in Assert-PSrule

### DISCUSSION THREAD

**Discussion by JonasCordsen** [Azure/PSRule.Rules.Azure#1541] [Q&A] [answered]
I would like to be able to use the public shared Baseline in local testing via Powershell

When I do local rule testing via the powershell module with the following command:
Assert-PSRule -format File -InputPath . -Baseline 'Azure.GA_2022_06' -Path ".\.ps-rule"

It seems like it is only processing the local rules I have created.
Is there support for this, am I doing it wrong, or is it on the roadmap for the powershell module?

**Comment by BernieWhite** [comment, marked as answer]
@JonasCordsen Thanks for the question. My understanding of your questions is that:

- You would like to run PSRule for Azure locally with the `Azure.GA_2022_06` baseline.

Is this correct?

---

To do this, you need to specify the `-Module` you want to use by default. For example:

```powershell
Assert-PSRule -Format File -InputPath . -Baseline 'Azure.GA_2022_06' -Path '.ps-rule/' -Module 'PSRule.Rules.Azure'
```

Alternatively you can configure `ps-rule.yaml` in the working directory, which is generally desired when you are working with Infrastructure as Code, but may not be applicable in your use case.

If you use `ps-rule.yaml` you can set the [Include.Module](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#includemodule) option.

---

If you have custom rules, then that you want to run as the same time as PSRule for Azure rules, then you should also set the [Rule.IncludeLocal](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#ruleincludelocal) option. This is because by default the baseline will exclude rules not included in PSRule for Azure.

---

I hope that helps.

**Reply by JonasCordsen** [reply]
Thanks @BernieWhite this was what I was looking for, the thing is we already had this in the ps-rule.yaml file, could it be that it does not see the file?

I found that if I specify the file with "-Option '.\ps-rule.yaml'" it works, I can continue to do this, but is there a better way/reason that it does not find it for It self?

**Reply by BernieWhite** [reply]
@JonasCordsen If `ps-rule.yaml` is in the same working path as where you are executing the cmdlet from it should be found. Especially if `-Option '.\ps-rule.yaml'` works.

A thing to note is that `ps-rule.yaml` is case-sensitive on Linux. i.e. it won' automatically find `PS-Rule.yaml`.

Otherwise not sure. I don't think this is currently logged in verbose logs so we might need to add that to work out what is occurring. 

You can also run `New-PSRuleOption -Verbose` from the same working path to troubleshoot what is being read from `ps-rule.yaml`.
